dashboard/test/models/user_level_test.rb

```diff
@@ -60,4 +58,40 @@ class UserTest < ActiveSupport::TestCase
     assert ul.finished?
     assert ul.passing?
   end
+
+  test "attempted, passed, and perfected scopes for nil result" do
+    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: nil)
+
+    assert UserLevel.count == 1
```

Comment:
Using `assert_equal 1, UserLevel.count` will give a better error message here
(something like "expected 1, got 2" instead of just "assertion failed").